src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java

```diff
@@ -41,33 +41,80 @@
+    private List<String> iosAdditionalSourceFiles = Arrays.asList("AppDelegate.m", "thread.m");
+
+    private static final List<String> javafxReflectionIosClassList = Arrays.asList(
+            "com.sun.prism.es2.ES2Pipeline",
+            "com.sun.prism.es2.IOSGLFactory",
+            "com.sun.javafx.font.coretext.CTFactory",
+            "com.sun.scenario.effect.impl.es2.ES2ShaderSource",
+            "com.sun.glass.ui.ios.IosApplication",
+            "com.sun.glass.ui.ios.IosCursor",
+            "com.sun.glass.ui.ios.IosGestureSupport",
+            "com.sun.glass.ui.ios.IosMenuBarDelegate",
+            "com.sun.glass.ui.ios.IosPlatformFactory",
+            "com.sun.glass.ui.ios.IosPixels",
+            "com.sun.glass.ui.ios.IosView",
+            "com.sun.glass.ui.ios.IosWindow"
+    );
+
+    private static final List<String>javafxJNIIosClassList = Arrays.asList(
+            "com.sun.glass.ui.ios.IosApplication",
+            "com.sun.glass.ui.ios.IosCursor",
+            "com.sun.glass.ui.ios.IosGestureSupport",
+            "com.sun.glass.ui.ios.IosMenuBarDelegate",
+            "com.sun.glass.ui.ios.IosPixels",
+            "com.sun.glass.ui.ios.IosView",
+            "com.sun.glass.ui.ios.IosWindow",
+            "java.util.Vector",
+            "com.sun.javafx.font.coretext.CGAffineTransform",
+            "com.sun.javafx.font.coretext.CGPoint",
+            "com.sun.javafx.font.coretext.CGRect",
+            "com.sun.javafx.font.coretext.CGSize",
+            "com.sun.javafx.font.FontConfigManager$FcCompFont",
+            "com.sun.javafx.font.FontConfigManager$FontConfigFont",
+            "com.sun.javafx.iio.ios.IosImageLoader"
+    );
+
+    private static final List<String> ioslibs = Arrays.asList(
+            "-lpthread", "-lz", "-lstrictmath", "-llibchelper",
+            "-ljava", "-ljvm", "-lnio", "-lzip", "-lnet");
+
+    private static final List<String> javafxLibs = Arrays.asList(
+            "prism_es2", "glass", "javafx_font", "prism_common", "javafx_iio");
 
     @Override
     List<String> getTargetSpecificLinkFlags(boolean useJavaFX, boolean usePrismSW) {
-        return Arrays.asList("-w", "-fPIC",
+        List<String> linkFlags = new ArrayList<>(Arrays.asList("-w", "-fPIC",
                 "-arch", Constants.ARCH_ARM64,
                 "-mios-version-min=11.0",
-                "-isysroot", getSysroot(),
-                "-Wl,-framework,Foundation",
-                "-Wl,-framework,UIKit",
-                "-Wl,-framework,CoreGraphics",
-                "-Wl,-framework,CoreText",
-                "-Wl,-framework,OpenGLES",
-                "-Wl,-framework,MobileCoreServices");
+                "-isysroot", getSysroot()));
+        if (useJavaFX) {
+            String javafxSDK = projectConfiguration.getJavafxStaticLibsPath().toString();
+            javafxLibs.forEach(name ->
+                    linkFlags.add("-Wl,-force_load," + javafxSDK + "/lib" + name + ".a"));
```

Comment:
At this point, we are guaranteed that the javafxStaticLibsPath exists, and is not null?
(projectConfiguration.getJavafxStaticLibsPath() always returns a valid path?)

Reply:
This is taken care by `FileDeps::setupDependencies` that is called before linking. 
I'll add an explicit verification of paths.

Reply:
I don't think that is needed at this place. More important is that the documentation of FileDeps::setupDependencies makes this very clear, and that it is also proven that that method has been called before this method is called.